test/config-luatest/gh_10304_allow_specific_function_access_test.lua

```diff
@@ -0,0 +1,70 @@
+local fun = require('fun')
+local yaml = require('yaml')
+local t = require('luatest')
+local treegen = require('test.treegen')
+local server = require('test.luatest_helpers.server')
+local helpers = require('test.config-luatest.helpers')
+
+local g = helpers.group()
+
+g.test_lua_call_func = function(g)
+    local dir = treegen.prepare_directory(g, {}, {})
+    local config = {
+        credentials = {
+            users = {
+                guest = {
+                    roles = {'super'},
+                },
+                alice = {
+                    password = "ALICE",
+                    privileges = {
+                        {
+                            permissions = {"execute"},
+                            lua_call = {"foo", "bar"}
```

Comment:
I would appreciate if we add test cases that verify `lua_call = {'all'}` and `lua_call = {'all', 'foo'}` scenarios. It is an existing functionality, but I'm not sure that it is properly tested (at least I remember that I had an intention to revisit it).

We can name the test in a more general way, say, `test/config-luatest/privileges_test.lua`, and add related test cases to it.

Reply:
added test cases, also added test with `config:reload`